feat: repository tutorial (#1778)
litestar-org/litestar: docs/tutorials/repository-tutorial/02-repository-introduction.rst

Interacting with repositories
-----------------------------

> Now that we've covered the modelling basics, we are able to create our first repository class.  The repository classes includes all of the standard CRUD operations as well as a few advanced features such as pagination, filtering and bulk operations.

Reviewer: ```suggestion
Now that we've covered the modeling basics, we are able to create our first repository class.  The repository classes includes all of the standard CRUD operations as well as a few advanced features such as pagination, filtering and bulk operations.
```

| ``remove``          | Removing Data  | Remove a single record from the database.                                                                                                                                                                                                   |
+---------------------+----------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| ``remove_many``     | Removing Data  | Remove one or more records from the database.                                                                                                                                                                                               |
+---------------------+----------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

.. note::

>     - All three of the bulk DML operations will leverage dialect specific enhancements to be as efficient as possible.  In addition to using efficient bulk inserts binds, the repository will optionally leverage the multi-row ``RETURNING`` support where possible.  The repository will automatically detect this support from the SQL Alchemy driver, so no additional interaction is required to enable this.

Reviewer: ```suggestion
    - All three of the bulk DML operations will leverage dialect-specific enhancements to be as efficient as possible.  In addition to using efficient bulk inserts binds, the repository will optionally leverage the multi-row ``RETURNING`` support where possible. The repository will automatically detect this support from the SQLAlchemy driver, so no additional interaction is required to enable this.
```

    - SQL engines generally have a limit to the number of elements that can be appended into an `IN` clause.  The repository operations will automatically break lists that exceed this limit into multiple queries that are concatenated together before return.  You do not need to account for this in your own code.

In the following examples, we'll cover a few ways that you can use the repository within your applications.

> Model Repository
> ^^^^^^^^^^^

Reviewer: Sphinx will be mad
```suggestion
Model Repository
^^^^^^^^^^^^^^^
```

> ^^^^^^^^^^^^^^^^^^^^^^^^^^

Reviewer: ```suggestion
^^^^^^^^^^^
```

    :emphasize-lines: 68-69
    :linenos:

3. **Paginated Data Selection**: Next, let's explore how to select multiple records with pagination. This functionality is useful for handling large amounts of data by breaking the data into manageable 'pages' or subsets.  ``LimitOffset`` is one of several filter types you can use with the repository.

.. literalinclude:: /examples/contrib/sqlalchemy/sqlalchemy_repository_bulk_operations.py
    :language: python
    :caption: app.py
    :emphasize-lines: 74-75
    :linenos:

4. **Bulk Delete**: Here we demonstrate how to perform a bulk delete operation. Just as with the bulk insert, deleting multiple records with the batch record methods is more efficient than executing row-by-row.

 .. literalinclude:: /examples/contrib/sqlalchemy/sqlalchemy_repository_bulk_operations.py
    :language: python
    :caption: app.py
    :emphasize-lines: 77-79
    :linenos:

5. **Counts**: Finally, We'll demonstrate how to count the number of records remaining in the database.

.. literalinclude:: /examples/contrib/sqlalchemy/sqlalchemy_repository_bulk_operations.py
    :language: python
    :caption: app.py
    :emphasize-lines: 81-83
    :linenos:

> Now that we have demonstrated how to interact with the repository objects outside of a Litestar application, our next example will use dependency injection to add this functionality to a Controller!

Reviewer: ```suggestion
Now that we have demonstrated how to interact with the repository objects outside of a Litestar application, our next example will use dependency injection to add this functionality to a Controller!
```

Link `Controller` here if capitalized with something like `:class:`Controller <.controller.Controller>` (not sure the exact path, not able to look atm)
